web/src/enterprise/repo/settings/RepoSettingsCodeIntelligencePage.tsx

```diff
@@ -84,6 +126,12 @@ export const RepoSettingsCodeIntelligencePage: FunctionComponent<Props> = ({ rep
     // State used in the toggle component shows or hides incomplete uploads
     const [onlyCompleted, setState] = useState(true)
 
+    // This observable emits values after successful deletion of an upload and
+    // force each filter connection to refresh. As these lists can share the
+    // same underlying entity, we need to refresh both at once on deletion of
+    // any upload.
+    const onDelete = useRef(new Subject<void>())
```

Comment:
This is not actually mutated, so `useRef` is not needed, see [examples](https://sourcegraph.com/search?q=r:github.com/sourcegraph/sourcegraph+lang:typescript+useMemo%28%28%29+%3D%3E+new+Subject&patternType=literal).

```suggestion
    const onDelete = useMemo(() => new Subject<void>())
```